client/client.go

```diff
@@ -272,56 +278,16 @@ func NewClient(svrAddrs []string, security SecurityOption, opts ...ClientOption)
 	return NewClientWithContext(context.Background(), svrAddrs, security, opts...)
 }
 
-// NewClientWithContext creates a PD client with context.
+// NewClientWithContext creates a PD client with context. This API uses the default keyspace id 0.
 func NewClientWithContext(ctx context.Context, svrAddrs []string, security SecurityOption, opts ...ClientOption) (Client, error) {
-	log.Info("[pd] create pd client with endpoints", zap.Strings("pd-address", svrAddrs))
-	c, clientCtx, clientCancel, tlsCfg := createClient(ctx, 0, &security)
-	// Inject the client options.
-	for _, opt := range opts {
-		opt(c)
-	}
-
-	c.svcDiscovery = newPDServiceDiscovery(clientCtx, clientCancel, &c.wg, addrsToUrls(svrAddrs), tlsCfg, c.option)
-	c.tsoClient = newTSOClient(clientCtx, clientCancel, &c.wg, c.option, c.keyspaceID, c.svcDiscovery, c.svcDiscovery.(tsoAllocatorEventSource), &pdTSOStreamBuilderFactory{})
-	if err := c.setup(); err != nil {
-		c.cancel()
-		return nil, err
-	}
-	if err := c.tsoClient.setup(); err != nil {
-		c.cancel()
-		return nil, err
-	}
-	return c, nil
+	// pass 0 as the default keyspace id
+	return NewClientWithKeyspace(ctx, 0, svrAddrs, security, opts...)
 }
 
-// NewTSOClientWithContext creates a TSO client with context.
-// TODO:
-// Merge NewClientWithContext with this API after we let client detect service mode provided on the server side.
-// Before that, internal tools call this function to use mcs service.
-func NewTSOClientWithContext(ctx context.Context, keyspaceID uint32, svrAddrs []string, security SecurityOption, opts ...ClientOption) (Client, error) {
-	log.Info("[tso] create tso client with endpoints", zap.Strings("pd(api)-address", svrAddrs))
-	c, clientCtx, clientCancel, tlsCfg := createClient(ctx, keyspaceID, &security)
-	// Inject the client options.
-	for _, opt := range opts {
-		opt(c)
-	}
-
-	c.svcDiscovery = newPDServiceDiscovery(clientCtx, clientCancel, &c.wg, addrsToUrls(svrAddrs), tlsCfg, c.option)
-	if err := c.setup(); err != nil {
-		c.cancel()
-		return nil, err
-	}
-
-	tsoSvcDiscovery := newTSOServiceDiscovery(clientCtx, clientCancel, &c.wg, MetaStorageClient(c), c.GetClusterID(c.ctx), keyspaceID, addrsToUrls(svrAddrs), tlsCfg, c.option)
-	c.tsoClient = newTSOClient(clientCtx, clientCancel, &c.wg, c.option, c.keyspaceID, tsoSvcDiscovery, tsoSvcDiscovery.(tsoAllocatorEventSource), &tsoTSOStreamBuilderFactory{})
-	if err := c.tsoClient.setup(); err != nil {
-		c.cancel()
-		return nil, err
-	}
-	return c, nil
-}
+// NewClientWithKeyspace creates a client with context and the specified keyspace id.
+func NewClientWithKeyspace(ctx context.Context, keyspaceID uint32, svrAddrs []string, security SecurityOption, opts ...ClientOption) (Client, error) {
+	log.Info("[pd] create pd client with endpoints", zap.Strings("pd-address", svrAddrs))
```

Comment:
What about logging the `keyspaceID` here?

Reply:
added logging for keyspaceID